core/issuance/issuance.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package issuance
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core/state/snapshot"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+// Issuance calculates the Ether issuance (or burn) across two state tries. In
+// normal mode of operation, the expectation is to calculate the issuance between
+// two consecutive blocks.
+func Issuance(block *types.Block, parent *types.Header, db *trie.Database, config *params.ChainConfig) (*big.Int, error) {
```

Comment:
IMO this is not issuance, but "EtherDelta". Issuance is "how much is issued", which IMO should be mining-rewards. 

This is how I think of the terms:
- `rewards`: mining rewards
- `burn`: 1559 burn
- `issuance`: `rewards` - `burn`
- `delta`: `issuance` - `destroyed ether`

And the `supply` is the `delta` between `empty-state` (before genesis) and  chain head.